### PR TITLE
1426 time and location check, refactored datetime utils and fixed a f…

### PR DESCRIPTION
**Pull request by hwindsor**
…ew bugs

## Pull Request Overview

- Validation of time and location fields on AGS creation
- Refactoring datetime utils to remove all the unused ones
- Major rework of the tests to send valid location IDs

---

### CODE REVIEW

File: app/modules/utils.py [hunk 1]
```diff
@@ -52,30 +52,21 @@ def is_valid_longitude(value):
     return min_longitude <= value <= max_longitude
 
 
-# this is a messy one. do we want to require timezone?  do we accept
-#   anything other than "full" ISO-8601?  for now just taking only full
-#   date + time (seconds optional) and ignoring timezone.  ymmv / caveat emptor / etc
+# Make this quite robust. Timezone is mandatory. Anything else is just making our life
```
Comment (drewblount) [inline app/modules/utils.py:55]: is this comment a TODO?

**Review by drewblount — APPROVED** [review]
LGTM

File: app/modules/utils.py (hunk 1 above)
Comment (hwindsor) [reply app/modules/utils.py:55]: It's not, but I can see why you think it is, will reword it.

**Review by naknomum — APPROVED** [review]
looks good.

**Pull request merged**